yajsapi/market/service.ts

```diff
@@ -100,7 +96,7 @@ export class MarketService {
     try {
       const { result: isProposalValid, reason } = await this.isProposalValid(proposal);
       if (isProposalValid) {
-        const chosenPlatform = this.getCommonPaymentPlatforms(proposal.properties)![0];
+        const chosenPlatform = this.allocation!.paymentPlatform;
```

Comment:
I know, but it's problematic. The allocation is created asynchronously so we can initialize it only in the async run method, so it must be of type `Allocation | undefined`. However, all of these private methods (which use this.allocation) are called after the run is executed, so there can't be a situation where the allocation will be undefined. To avoid using.`this.allocation!` we can add a condition: `if (!this.allocation) throw new Error('The service has not been started correctly')` but it will never happen... so i don't know which is better, but i think I'll add a condition